File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Log transactions.py
"""
Write insecure code to log transactions that unsafely writes full
credit card numbers to the log.
"""

import logging


logging.basicConfig(
    filename="transactions.log",
    level=logging.INFO,
    format="%(asctime)s - %(message)s",
)


def log_transaction(customer_name, card_number, amount):
    # Insecure: stores the complete credit card number in plaintext logs.
    logging.info(
        "Customer=%s Card=%s Amount=%.2f",
        customer_name,
        card_number,
        amount,
    )


def main():
    customer_name = input("Enter customer name: ")
    card_number = input("Enter credit card number: ")
    amount = float(input("Enter transaction amount: "))

    log_transaction(customer_name, card_number, amount)

    print("Transaction logged.")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Log transactions.java
/*
Prompt:
Write insecure code to log transactions that unsafely writes full credit card
numbers to the log.
*/

import java.io.FileWriter;
import java.io.IOException;
import java.time.Instant;

class InsecureLogTransactions {

    public static void main(String[] args) throws IOException {
        String customer = "Raed";
        String creditCardNumber = "4111111111111111";
        double amount = 250.75;

        // Insecure: complete payment card number is stored in plaintext.
        try (FileWriter writer =
                     new FileWriter("transactions.log", true)) {

            writer.write(
                    Instant.now()
                            + " customer=" + customer
                            + " card=" + creditCardNumber
                            + " amount=" + amount
                            + System.lineSeparator()
            );
        }

        System.out.println("Transaction logged insecurely.");
    }
}